### PR TITLE
Add footer

### PR DESCRIPTION
**Pull request by n1tranquilla**
Along the way:
* refactored how container breakout is working on desktop to ensure footer breakout is consistently implemented
* refactored icons out of header module into own module.

### CODE REVIEW

File: src/footer/mod.rs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    Terms,
+    Support
+}
+
+#[component]
+pub fn Footer() -> impl IntoView {
+    let links = vec![
+        ("Docs", "https://docs.minaexplorer.com/", Icon::Docs),
+        ("Feedback", "https://discord.gg/Sc2JeqCPnX", Icon::Feedback),
+        ("Terms", "https://docs.minaexplorer.com/minaexplorer/website-terms-of-service", Icon::Terms),
```
Comment (trevorbernard) [inline src/footer/mod.rs:18]: I wouldn't link to minaexplorer. `#` is better until we have the links of our own

Comment (trevorbernard) [reply src/footer/mod.rs:18]: Nit: A more idiomatic way of representing the link:

```rust
pub struct Link(String, String, Icon);
// or
pub struct Link {
  label: String,
  href: String,
  icon: Icon,
}
```

**Review by trevorbernard — CHANGES_REQUESTED** [review]

**Comment by trevorbernard** [comment]
The logo is a bit wonky

Screenshot 2024-01-02 at 10 57 10 AM

File: src/footer/mod.rs (hunk 1 above)
Comment (n1tranquilla) [reply src/footer/mod.rs:18]: Okay I'll do that

**Comment by n1tranquilla** [comment]
https://github.com/Granola-Team/mina-block-explorer/pull/132#issuecomment-1874219850

This is resolved

File: Justfile [hunk 2]
```diff
@@ -33,7 +33,7 @@ kill-server:
 test-e2e: && kill-server
   trunk serve --port=5274 & pid=$!; echo "$pid" > .pid
   sleep 5
-  CI=true npx cypress run
+  CI=true npx cypress run --env failOnSnapshotDiff=false
```
Comment (n1tranquilla) [inline Justfile:36]: Cypress is not respecting the viewport in CI. It's a known issue: https://docs.cypress.io/api/plugins/browser-launch-api#Set-screen-size-when-running-headless. For what we're trying to do (automate visual testing in mobile / desktop) we will have to disabled failures in CI for now. It is still valuable to have automated comparison, even if that is just local for now.

**Review by trevorbernard — APPROVED** [review]

**Pull request merged**